daft/io/common.py

```diff
@@ -55,38 +56,45 @@ def _get_tabular_files_scan(
                 path,
                 file_format_config,
                 storage_config,
-                schema=schema_hint._schema if schema_hint is not None else None,
+                schema_hint=schema_hint._schema if schema_hint is not None else None,
             )
         elif isinstance(path, str):
             scan_op = ScanOperatorHandle.glob_scan(
                 [path],
                 file_format_config,
                 storage_config,
-                schema=schema_hint._schema if schema_hint is not None else None,
+                schema_hint=schema_hint._schema if schema_hint is not None else None,
             )
         else:
             raise NotImplementedError(f"_get_tabular_files_scan cannot construct ScanOperatorHandle for input: {path}")
 
         builder = LogicalPlanBuilder.from_tabular_scan_with_scan_operator(
             scan_operator=scan_op,
-            schema_hint=schema_hint,
         )
         return builder
 
     paths = path if isinstance(path, list) else [str(path)]
     runner_io = get_context().runner().runner_io()
     file_infos = runner_io.glob_paths_details(paths, file_format_config=file_format_config, io_config=io_config)
 
-    # Infer schema if no hints provided
-    inferred_or_provided_schema = (
-        schema_hint
-        if schema_hint is not None
-        else runner_io.get_schema_from_first_filepath(file_infos, file_format_config, storage_config)
-    )
+    # Infer schema
+    schema = runner_io.get_schema_from_first_filepath(file_infos, file_format_config, storage_config)
+
+    # Apply hints from schema_hints if provided
+    if schema_hint is not None:
+        # If CSV and no headers, then use the schema hint as the schema
+        if isinstance(file_format_config.config, CsvSourceConfig) and file_format_config.config.has_headers == False:
+            if len(schema) != len(schema_hint):
+                raise ValueError(
```

Comment:
Based on this existing test case https://github.com/Eventual-Inc/Daft/blob/06c2ccf920fb32796a891c35ecc9f3c304c921b1/tests/dataframe/test_creation.py#L484-L511

I thought maybe it makes sense for schema_hints to be the definitive schema when csv has no headers, as a way to provide named columns instead of the default "column_0" or "column_1", and this would only work if hints for all columns are provided. 

But I also agree that it would be simpler and consistent to remove this invariant and let the user realize that column names will default to "column_0" etc., and they can rename their schema hints accordingly. and I also realize that column names can be changed with `.alias` 😅 

removed these checks in latest commit